pkg/build_zip.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This tool builds zip files from a list of inputs."""
+
+import sys
+
+from absl import flags
+from helpers import UnquoteAndSplit
+from zipfile import ZipFile, ZipInfo, ZIP_DEFLATED
+
+flags.DEFINE_string('output', None, 'The output file, mandatory')
+flags.mark_flag_as_required('output')
+
+flags.DEFINE_multi_string('file', [], 'A file to add to the zip')
+
+FLAGS = flags.FLAGS
+
+def main(unused_argv):
+  with ZipFile(FLAGS.output, 'w') as zip:
+    for f in FLAGS.file:
+      (src_path, dst_path) = UnquoteAndSplit(f, '=')
+
+      # remove leading /'s: the zip format spec says paths should never have a
+      # leading slash, but Python will happily do this. The built-in zip tool
+      # in Windows will complain that such a zip file is invalid.
+      if dst_path[0] == '/':
+          dst_path = dst_path[1:]
```

Comment:
Maybe this is weird to have in `build_zip.py`. I'm open to alternatives. My thought was that (1) its definitely a compatibility problem to have leading slashes, but (2) thats a zip specific thing, and (3) doing it in Starlark would make this binary less flexible/would either be the same hack or more work (not sharing as much code with `pkg_tar`.

Reply:
This seems reasonable for now.